doc/netplan-tutorial.md

```diff
@@ -0,0 +1,560 @@
+
+# Getting started with Netplan
+
+In this tutorial we'll drive you through a number of practical exercises to get you started with Netplan. By the end of the tutorial you will be familiarized with some of the Netplan commands and configuration.
+
+## Pre-requisites
+
+In order to do the exercises yourself you will need a virtual machine, preferably running Ubuntu. In this tutorial, we will use LXD to create virtual networks and launch virtual machines. Feel free to use a cloud instance or a different hypervisor. As long as you can achieve the same results, you should be fine. If you're going to use your own desktop/laptop system, some of the exercises might interrupt your network connectivity.
+
+If you already have a setup where you can do the exercises you can just skip this section.
+
+### Setting up the environment
+
+You can follow the steps below to install and create a basic LXD configuration you can use to launch virtual machines. For more information about LXD, please visit: https://linuxcontainers.org/lxd/introduction/
+
+If you don't have LXD installed in your system, follow the steps below to install it:
+
+```bash
+$ snap install lxd
+$ lxd init --minimal
+```
+
+For some of the exercises you will need a second network interface in your virtual machine. Run the command below to create a new network in LXD:
+
+```bash
+lxc network create netplanbr0 --type=bridge
+```
+
+You should see the output below:
+
+```
+Network netplanbr0 created
+```
+
+After executing the commands above you will have a usable LXD installation with a working network bridge.
+
+Now create a virtual machine called `netplan-lab0`:
+
+```bash
+lxc init --vm ubuntu:22.04 netplan-lab0
```

Comment:
I wonder if we should use something newer here, like `ubuntu-daily:devel` to make the latest Netplan features (like "status" CLI) available.

Reply:
hmm I don't know, I think the focus should be stable releases, those are the ones users will install after all. We can update it to ubuntu:23.04 in a couple of months I guess. For now I think I will mention netplan status is only available in netplan 0.106.